app/controllers/charge_controller.js

```diff
@@ -212,7 +212,22 @@ module.exports = {
         redirect(res).toAuth3dsRequired(req.chargeId)
         break
       default:
-        redirect(res).toConfirm(req.chargeId)
+        if (charge.walletType !== undefined) {
```

Comment:
My JS is a bit rusty but if `charge.walletType` is `null` it'll pass this check won't it?
```
> null !== undefined
true
```
 I ask because Connector should always return `wallet_type`, it'll be `null` for non-wallet payments else `GOOGLE_PAY || APPLE_PAY`.  I think we need to check that it isn't a) there but set to null b) not there at all (just to be safe). I guess it depends how Connector response is deserialised and if `wallet_type: null` and `wallet_type` not existing at all becomes `walletType = undefined` on the charge object 🤷‍♂️ .

Reply:
Actually our current `ChargeResponse` does not include null values, those keys will be omitted. Perhaps we should follow this pattern (recommended by google style guide). In that case `wallet_type` will only be present if its a wallet payment with values `APPLE_PAY | GOOGLE_PAY`. Probably still need to consider the scenario where the key exists and its value is `null` since if our design changes to include `null` values we don't want to accidentally skip confirm page for card payments.

Reply:
I think this is covered by 
https://github.com/alphagov/pay-frontend/blob/28c643cd5cb0007adc280ba2a0f7c3d69a90fa1c/app/services/normalise_charge.js#L35-L37

Reply:
The normailiser will omit it, so if it’s `null` it will always be `undefined` as the controller sees it

Reply:
ah cool 👍